geo/geomfn: `TestAngle{,LineString}` fail on s390x

**[How to gain access to an s390x machine for testing (Internal link)](https://cockroachlabs.atlassian.net/wiki/spaces/devinf/pages/4536467683/Reproducing+an+issue+on+s390x)**

```
--- FAIL: TestAngle (0.00s)
    --- FAIL: TestAngle/POINT_(0_0),_POINT_(0_1),_POINT_(0_0),_POINT_(1_1) (0.00s)
        angle_test.go:61:
                Error Trace:    pkg/geo/geomfn/angle_test.go:61
                Error:          Not equal:
                                expected: 0.7853981633974483
                                actual  : 0.7853981633974484
                Test:           TestAngle/POINT_(0_0),_POINT_(0_1),_POINT_(0_0),_POINT_(1_1)
    --- FAIL: TestAngle/POINT_(10_10),_POINT_(10_11),_POINT_(-100_-100),_POINT_(-99_-99) (0.00s)
        angle_test.go:61:
                Error Trace:    pkg/geo/geomfn/angle_test.go:61
                Error:          Not equal:
                                expected: 0.7853981633974483
                                actual  : 0.7853981633974484
                Test:           TestAngle/POINT_(10_10),_POINT_(10_11),_POINT_(-100_-100),_POINT_(-99_-99)
--- FAIL: TestAngleLineString (0.00s)
    --- FAIL: TestAngleLineString/LINESTRING_(0_0,_0_1),_LINESTRING_(0_0,_1_1) (0.00s)
	angle_test.go:137:
		Error Trace:    pkg/geo/geomfn/angle_test.go:137
		Error:          Not equal:
				expected: 0.7853981633974483
				actual  : 0.7853981633974484
		Test:           TestAngleLineString/LINESTRING_(0_0,_0_1),_LINESTRING_(0_0,_1_1)
    --- FAIL: TestAngleLineString/LINESTRING_(10_10,_10_11),_LINESTRING_(-100_-100,_-99_-99) (0.00s)
	angle_test.go:137:
		Error Trace:    pkg/geo/geomfn/angle_test.go:137
		Error:          Not equal:
				expected: 0.7853981633974483
				actual  : 0.7853981633974484
		Test:           TestAngleLineString/LINESTRING_(10_10,_10_11),_LINESTRING_(-100_-100,_-99_-99)
```

Jira issue: CRDB-50316

Epic CRDB-21133

--- a/pkg/geo/geomfn/BUILD.bazel
+++ b/pkg/geo/geomfn/BUILD.bazel
@@ -111,6 +111,7 @@ go_test(
         "//pkg/geo/geopb",
         "//pkg/geo/geos",
         "//pkg/geo/geotest",
+        "//pkg/testutils/floatcmp",
         "@com_github_cockroachdb_errors//:errors",
         "@com_github_stretchr_testify//assert",
         "@com_github_stretchr_testify//require",

--- a/pkg/geo/geomfn/angle_test.go
+++ b/pkg/geo/geomfn/angle_test.go
@@ -8,12 +8,31 @@ package geomfn
 import (
 	"fmt"
 	"math"
+	"runtime"
+	"strconv"
 	"testing"
 
 	"github.com/cockroachdb/cockroach/pkg/geo"
 	"github.com/cockroachdb/cockroach/pkg/geo/geopb"
+	"github.com/cockroachdb/cockroach/pkg/testutils/floatcmp"
 	"github.com/stretchr/testify/require"
 )
+
+func assertFloatsMatch(t *testing.T, expected, actual float64) {
+	if runtime.GOARCH != "s390x" {
+		require.Equal(t, expected, actual)
+		return
+	}
+	// s390x system has architecture support for 'atan' function (which is used
+	// in 'angleFromCoords') and produces slightly different result than math
+	// operations we do to get the expected number. To go around this, we use
+	// the custom float comparison that checks 15 significant decimal digits.
+	exp := strconv.FormatFloat(expected, 'f', -1, 64)
+	act := strconv.FormatFloat(actual, 'f', -1, 64)
+	matched, err := floatcmp.FloatsMatch(exp, act)
+	require.NoError(t, err)
+	require.True(t, matched)
+}
 
 func TestAngle(t *testing.T) {
 	pf := func(f float64) *float64 {
@@ -58,7 +77,7 @@ func TestAngle(t *testing.T) {
 			angle, err := Angle(g1, g2, g3, g4)
 			require.NoError(t, err)
 			if tc.expected != nil && angle != nil {
-				require.Equal(t, *tc.expected, *angle)
+				assertFloatsMatch(t, *tc.expected, *angle)
 			} else {
 				require.Equal(t, tc.expected, angle)
 			}
@@ -134,7 +153,7 @@ func TestAngleLineString(t *testing.T) {
 			angle, err := AngleLineString(g1, g2)
 			require.NoError(t, err)
 			if tc.expected != nil && angle != nil {
-				require.Equal(t, *tc.expected, *angle)
+				assertFloatsMatch(t, *tc.expected, *angle)
 			} else {
 				require.Equal(t, tc.expected, angle)
 			}

